### DISCUSSION TITLE
ModuleNotFoundError when opening a macOS bundle that uses external libraries

### DISCUSSION THREAD

**Discussion by AnshThaker** [flet-dev/flet#2596] [Q&A] [answered]
### Question

I have created a flet app that uses a bunch of libraries. The libraries are:
1. flet (of course)
2. google.auth.exceptions
3. gspread
4. google.auth2.service_account

After creating a macOS bundle for the app using `flet build macos`, when I open the app it shows a ModuleNotFoundError.

![image](https://github.com/flet-dev/flet/assets/102654162/efeea53b-384e-4f69-b084-8283e08cc6ce)

### Error message

```bash
Error running app
Traceback (most recent call last):
  File "<string>", line 40, in <module>
  File "<frozen runpy>", line 229, in run_module
  File "<frozen runpy>", line 88, in _run_code
  File "/var/folders/0g/5p3l1qvd29z7118hdq5vs9sr0000gn/T/serious_python_temp1BxhTK/main.py", line 2, in <module>
ModuleNotFoundError: No module named 'google'
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by FeodorFitsner** [comment, marked as answer]
Do you have those modules in your `requirements.txt`?

**Reply by AnshThaker** [reply]
Thanks, I hadn't added the modules to `requirements.txt`. Do I need to do the same if I'm planning to build for Android?

**Reply by kentbull** [reply]
Since I use Pipenv I found I had to generate a `requirements.txt` file with `pipenv requirements > requirements.txt` in order to get `flet build macos` to properly recognize the needed dependencies for my app. And, I had to remove the index URL line at the top of the file:
```
-i https://pypi.org/simple
```

For some reason that was throwing the Flet build tool off.

After making these changes I was able to use `flet macos build` (with a few parameters) and successfully generate an app I could use on MacOS.

Full command seen below:
```bash
flet build -o ./build macos src/myapp
```

From pyproject.toml
```toml
[tool.flet]
entry_point = "src/myapp/main.py"
```

**Reply by kentbull** [reply]
For whatever reason this no longer works for me now that I use Poetry.
